sycl/include/sycl/ext/intel/math/imf_integer_utils.hpp

```diff
@@ -0,0 +1,144 @@
+//==--------------------------- imf_integer_utils.hpp ----------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// C++ APIs for simple integer util funcitons
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+extern "C" {
+unsigned __imf_brev(unsigned);
+unsigned long long __imf_brevll(unsigned long long);
+unsigned __imf_byte_perm(unsigned, unsigned, unsigned);
+long long __imf_llmax(long long x, long long y);
+long long __imf_llmin(long long x, long long y);
+int __imf_max(int x, int y);
+int __imf_min(int x, int y);
+unsigned long long __imf_ullmax(unsigned long long x, unsigned long long y);
+unsigned long long __imf_ullmin(unsigned long long x, unsigned long long y);
+unsigned __imf_umax(unsigned x, unsigned y);
+unsigned __imf_umin(unsigned x, unsigned y);
+int __imf_clz(int);
+int __imf_clzll(long long);
+int __imf_ffs(int);
+int __imf_ffsll(long long);
+int __imf_mul24(int, int);
+int __imf_mulhi(int, int);
+long long __imf_mul64hi(long long, long long);
+int __imf_popc(unsigned);
+int __imf_popcll(unsigned long long);
+int __imf_rhadd(int, int);
+int __imf_hadd(int, int);
+unsigned __imf_sad(int, int, unsigned);
+unsigned __imf_uhadd(unsigned, unsigned);
+unsigned __imf_umul24(unsigned, unsigned);
+unsigned __imf_umulhi(unsigned, unsigned);
+unsigned long long __imf_umul64hi(unsigned long long, unsigned long long);
+unsigned __imf_urhadd(unsigned, unsigned);
+unsigned __imf_usad(unsigned, unsigned, unsigned);
+}
+
+namespace sycl {
+inline namespace _V1 {
+namespace ext::intel::math {
+
+template <typename Tp = unsigned> Tp brev(Tp x) { return __imf_brev(x); }
```

Comment:
Currently nothing is stopping the user from passing a `Tp` that is implicitly convertible to `unsigned` here, but that does not mean that `Tp` can be constructed from `unsigned`. The best solution I can think of is
```suggestion
template <typename Tp = unsigned> unsigned brev(Tp x) { return __imf_brev(x); }
```
This applies to all the other functions as well.

Reply:
Hi, @steffenlarsen 
Done.
Thanks very much.